lib/inertia_rails/renderer.rb

```diff
@@ -74,25 +74,21 @@ def merge_props(shared_data, props)
     end
 
     def computed_props
-      _props = merge_props(shared_data, props).select do |key, prop|
-        if rendering_partial_component?
-          partial_keys.none? || key.in?(partial_keys) || prop.is_a?(AlwaysProp)
-        else
-          !prop.is_a?(LazyProp)
-        end
-      end
+      _props = merge_props(shared_data, props)
 
-      drop_partial_except_keys(_props) if rendering_partial_component?
+      deep_transform_props _props do |prop, path|
+        next [:dont_keep] unless keep_prop?(prop, path)
```

Comment:
nitpick: could we move these symbol values to constants since they're acting as an enum?

Reply:
sure! as an aside, `:dont_keep` doesn't do anything at all... it's just there to reveal intention